src/bidiMapper/domains/session/EventManager.ts

```diff
@@ -188,17 +192,31 @@ export class EventManager extends EventEmitter<EventManagerEventsMap> {
         }
       }
     }
+
+    await this.toggleModulesIfNeeded();
```

Comment:
This should be done before `this.#subscriptionManager.subscribe`. Otherwise the events will start coming before the subscription is done.

Reply:
Hmm I guess so but that makes the current implementation not work as it relies on the Maps in SessionManager.

Reply:
> Hmm I guess so but that makes the current implementation not work as it relies on the Maps in SessionManager.

Right. Then we can try this approach.